From: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Date: Tue, 15 May 2012 09:18:55 +0000
Subject: [PATCH 35/51] e1000: look into the page instead of skb->data for
 e1000_tbi_adjust_stats()
Patch-mainline: v3.5-rc1
Git-commit: 281a8f2462fe3cd2395902955205621e2c519464
References: bnc#795304 FATE#313649

This is another fixup where the data is not transfered into buffer
addressed by skb->data but into a page.

Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Acked-by: John Ronciak <john.ronciak@intel.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/intel/e1000/e1000_main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/intel/e1000/e1000_main.c
+++ b/drivers/net/ethernet/intel/e1000/e1000_main.c
@@ -4061,7 +4061,7 @@ static bool e1000_clean_jumbo_rx_irq(str
 				spin_lock_irqsave(&adapter->stats_lock,
 				                  irq_flags);
 				e1000_tbi_adjust_stats(hw, &adapter->stats,
-				                       length, skb->data);
+						       length, mapped);
 				spin_unlock_irqrestore(&adapter->stats_lock,
 				                       irq_flags);
 				length--;
